### PR TITLE
Add some more info to the Jenkins README.

### PR DESCRIPTION
**Pull request by spxtr**
This is a bit of a work-in-progress, and I'd appreciate feedback on what to add or remove. I'm not sure that I need to say so much about the GCS format, and I should probably say some more about JJB.

@kubernetes/sig-testing

### CODE REVIEW

File: hack/jenkins/README.md [hunk 1]
```diff
@@ -41,6 +41,39 @@ outside this repository, it's tricky to keep documentation for it up
 to date quickly. However, the scripts themselves attempt to provide
 color for the configuration(s) that each script runs in.
 
+## GCS Log Format
+
+Our `upload-to-gcs.sh` script runs at the start and end of every job. Logs on
+post-commit Jenkins go under `gs://kubernetes-jenkins/logs/`. Logs on PR
+Jenkins go under `gs://kubernetes-jenkins-pull/pr-logs/pull/PULL_NUMBER/`.
```
Comment (spiffxp) [inline hack/jenkins/README.md]: are you talking about separate jenkins instances or separate triggers here?

might be worth a sketch of how these are different (eg: which jobs run post-commit vs per PR)

File: hack/jenkins/README.md [hunk 2]
```diff
@@ -41,6 +41,39 @@ outside this repository, it's tricky to keep documentation for it up
...
+Individual run logs go into the `JOB_NAME/BUILD_NUMBER` folder.
+
+At the start of the job, it uploads `started.json` containing the version and
```
Comment (spiffxp) [inline hack/jenkins/README.md]: version of... kubernetes under test?

File: hack/jenkins/README.md [hunk 3]
```diff
@@ -41,6 +41,39 @@ outside this repository, it's tricky to keep documentation for it up
...
+  12345/
+    build-log.txt
+    started.json
+    finished.json
+    artifacts/
+      gcp-resources-{before, after}.txt
+      junit_{00, 01, ...}.xml
+      jenkins-e2e-master/{kube-apiserver.log, ...}
+      jenkins-e2e-minion-abcd/{kubelet.log, ...}
+  12344/
```
Comment (spiffxp) [inline hack/jenkins/README.md]: :+1: might be worth calling out which of these are required for munger to do its thing?

**Comment by spiffxp** [comment]
kubernetes-e2e-aws is listed as running every half-hour

**Comment by spiffxp** [comment]
Can probably address #24127 here as well

File: hack/jenkins/README.md (hunk 1 above)
Comment (spxtr) [inline hack/jenkins/README.md]: Added that above.

File: hack/jenkins/README.md (hunk 2 above)
Comment (spxtr) [inline hack/jenkins/README.md]: Done.

File: hack/jenkins/README.md (hunk 3 above)
Comment (spxtr) [inline hack/jenkins/README.md]: Done. This is subject to change if it starts looking at started/finished.json.

**Comment by spxtr** [comment]
Changed kubernetes-e2e-aws' line to indicate its frequency.

@spiffxp thanks for your comments, PTAL.

**Comment by spiffxp** [comment]
LGTM

**Comment by ixdy** [comment]
LGTM

**Pull request merged**